src/htm/algorithms/TemporalMemory.cpp

```diff
@@ -185,45 +185,26 @@ void TemporalMemory::growSynapses_(
 			 const Segment& segment,
                          const SynapseIdx nDesiredNewSynapses,
                          const vector<CellIdx> &prevWinnerCells) {
-  // It's possible to optimize this, swapping candidates to the end as
-  // they're used. But this is awkward to mimic in other
-  // implementations, especially because it requires iterating over
-  // the existing synapses in a particular order.
-
+  
   vector<CellIdx> candidates(prevWinnerCells.begin(), prevWinnerCells.end());
   NTA_ASSERT(std::is_sorted(candidates.begin(), candidates.end()));
 
-  // Skip cells that are already synapsed on by this segment
-  // Biological motivation (?):
-  // There are structural constraints on the shapes of axons & synapses 
-  // which prevent a large number duplicate of connections.
-  //
-  // It's important to prevent cells from growing duplicate synapses onto a segment, 
-  // because otherwise a strong input would be sampled many times and grow many synapses.
-  // That would give such input a stronger connection. 
-  // Synapses are supposed to have binary effects (0 or 1) but duplicate synapses give 
-  // them (synapses 0/1) varying levels of strength.
-  for (const Synapse& synapse : connections.synapsesForSegment(segment)) {
-    const CellIdx presynapticCell = connections.dataForSynapse(synapse).presynapticCell;
-    const auto already = std::lower_bound(candidates.cbegin(), candidates.cend(), presynapticCell);
```

Comment:
FIX: there was a bug here too.. if there are multiple (2) candidates to the same presynaptic cell (say 123), the "already" condition is not met. So we end up with a segment with duplicit synapses (no good). 
This is likely the reason for changed deterministic representations. 
FYI @ctrl-z-9000-times

Reply:
But the candidates list is a copy of `prevWinnerCells` which should be sorted & unique, no duplicate cells...

Reply:
>  candidates list is a copy of prevWinnerCells which should be sorted & unique, no duplicate cells...

you're right, the method `growSynapses_()` is only used internally, so we can verify it only ever gets the correct inputs. So this is not an issue.